### PR TITLE
Extract localizable strings from dotnet-new

### PR DESCRIPTION
**Pull request by piotrpMSFT**
@krwq @livarcocc @jgoshi @nguerrera 

This is a prototype for how we will extract localizable strings from existing product code. I plan to use this as a template for folks who will help us do the full extraction. Focus on design, naming, usability... this pattern will be all over the place.

### CODE REVIEW

File: src/dotnet/commands/dotnet-new/LocalizableStrings.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
+namespace Microsoft.DotNet.Tools.New
+{
+    internal static class LocalizableStrings
+    {
+        public static const string AppName = "dotnet new";
+
```
Comment (krwq) [inline src/dotnet/commands/dotnet-new/LocalizableStrings.cs]: do we need the extra spaces? I don't think they make it easier to read

File: src/dotnet/commands/dotnet-new/LocalizableStrings.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        public static const string AppFullName = ".NET Initializer";
+
+        public static const string AppDescription = "Initializes empty project for .NET Platform"; 
+
+        public static const string CreateFailedBecauseProjectExists = "Creating new {0} project failed, project already exists.";
+
+        public static const string CreateFailedBecauseContentExists = "Creating new {0} project failed, directory already contains {1}";
+
+        public static const string CreateFailed = "Creating new {0} project failed.";
```
Comment (krwq) [inline src/dotnet/commands/dotnet-new/LocalizableStrings.cs]: When I look only on this string I believe it is hard to read the context how to translate this sentence (I wouldn't know how to translate this to Polish without more context or some example word I can put there) - I think it should be something like {ProgrammingLanguageName} or at least a comment explaining: {0} -> i.e. "C#"

Comment (piotrpMSFT) [inline src/dotnet/commands/dotnet-new/LocalizableStrings.cs]: We'll get into the nitt-gritty of translation and adjust. For now we are focusing on the calling convention from CLI into the localization objects. I agree we'll likely need to give help, but it's unclear what that format will be.

File: src/dotnet/commands/dotnet-new/LocalizableStrings.cs (hunk 1 above)
Comment (piotrpMSFT) [inline src/dotnet/commands/dotnet-new/LocalizableStrings.cs]: I don't want to change style subjectively. Distinct statements get a line in between to show they're not meaningfully related. Consistency avoids subjective fights over "I don't think these are related!"

File: src/dotnet/commands/dotnet-new/LocalizableStrings.cs (hunk 2 above)
Comment (nguerrera) [inline src/dotnet/commands/dotnet-new/LocalizableStrings.cs]: The loc team will file bugs if they need clarification.

File: src/dotnet/commands/dotnet-new/LocalizableStrings.cs (hunk 1 above)
Comment (nguerrera) [inline src/dotnet/commands/dotnet-new/LocalizableStrings.cs]: In what context is "dotnet new" localizable. Isn't it what I always have to type?

Comment (nguerrera) [inline src/dotnet/commands/dotnet-new/LocalizableStrings.cs]: Also, I'm sure you've seen the build errors, but it should be just `const` not `static const` ;)

**Comment by nguerrera** [comment]
:+1:

**Comment by piotrpMSFT** [comment]
I'll close this PR at the end of the day, once everyone has had a chance to comment. Will then come back with the `Real` PR.

**Pull request closed without merging**